src/Marten/Events/Daemon/HighWater/HighWaterStatistics.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+
+namespace Marten.Events.Daemon.HighWater
+{
+    internal class HighWaterStatistics
+    {
+        public long LastMark { get; set; }
+        public long HighestSequence { get; set; }
+        public long CurrentMark { get; set; }
+        public bool HasChanged => CurrentMark > LastMark;
+        public DateTimeOffset? LastUpdated { get; set; }
+        public long SafeStartMark { get; set; }
+        public DateTimeOffset Timestamp { get; set; }
+
+        public HighWaterStatus InterpretStatus(HighWaterStatistics previous)
+        {
+            if (HighestSequence == 1 && CurrentMark == 0) return HighWaterStatus.CaughtUp;
```

Comment:
**QUESTION**: @jeremydmiller is it because HighestSequence numbered from 1 in database?

Reply:
Yep. Confused me a little bit to be honest, but Postgresql sequences start w/ 1 by default. So the initial state is "HighestSequence = 1", which is really the current value of the postgres sequence.